classes/controlledVocab/ControlledVocabEntry.php

```diff
@@ -3,101 +3,199 @@
+use PKP\core\traits\ModelWithSettings;
+use PKP\controlledVocab\ControlledVocab;
+use PKP\controlledVocab\ControlledVocabEntryMatch;
+
+class ControlledVocabEntry extends Model
 {
-    //
-    // Get/set methods
-    //
+    use ModelWithSettings;
+    
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Model::$table
+     */
+    protected $table = 'controlled_vocab_entries';
 
     /**
-     * Get the ID of the controlled vocab.
-     *
-     * @return int
+     * @copydoc \Illuminate\Database\Eloquent\Model::$primaryKey
      */
-    public function getControlledVocabId()
+    protected $primaryKey = 'controlled_vocab_entry_id';
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Concerns\GuardsAttributes::$guarded
+     */
+    protected $guarded = [
+        'controlledVocabEntryId',
+    ];
+
+    /**
+     * @copydoc \Illuminate\Database\Eloquent\Concerns\HasTimestamps::$timestamps
+     */
+    public $timestamps = false;
+
+    /**
+     * @copydoc \PKP\core\traits\ModelWithSettings::getSettingsTable
+     */
+    public function getSettingsTable(): string
     {
-        return $this->getData('controlledVocabId');
+        return 'controlled_vocab_entry_settings';
     }
 
     /**
-     * Set the ID of the controlled vocab.
-     *
-     * @param int $controlledVocabId
+     * @copydoc \Illuminate\Database\Eloquent\Concerns\HasAttributes::casts
      */
-    public function setControlledVocabId($controlledVocabId)
+    protected function casts(): array
     {
-        $this->setData('controlledVocabId', $controlledVocabId);
+        return [
+            'controlled_vocab_entry_id' => 'integer',
+            'controlled_vocab_id' => 'integer',
+            'seq' => 'float',
+        ];
     }
 
     /**
-     * Get sequence number.
-     *
-     * @return float
+     * @copydoc \PKP\core\traits\ModelWithSettings::getSchemaName
      */
-    public function getSequence()
+    public static function getSchemaName(): ?string
     {
-        return $this->getData('sequence');
+        return null;
     }
 
     /**
-     * Set sequence number.
-     *
-     * @param float $sequence
+     * @copydoc \PKP\core\traits\ModelWithSettings::getMultilingualProps
      */
-    public function setSequence($sequence)
+    public function getMultilingualProps(): array
     {
-        $this->setData('sequence', $sequence);
+        return array_merge(
+            $this->multilingualProps,
+            ControlledVocab::getDefinedVocabSymbolic(),
+            Arr::wrap(UserInterest::CONTROLLED_VOCAB_INTEREST)
+        );
     }
 
     /**
-     * Get the localized name.
-     *
-     * @return string
+     * @copydoc \PKP\core\traits\ModelWithSettings::getSettings
      */
-    public function getLocalizedName()
+    public function getSettings(): array
     {
-        return $this->getLocalizedData('name');
+        return array_merge(
+            $this->settings,
+            ControlledVocab::getDefinedVocabSymbolic(),
```

Comment:
As above -- mixing symbolic names and setting names?

Reply:
please see https://github.com/pkp/pkp-lib/pull/10324#discussion_r1888497825 .